### DISCUSSION TITLE
Duplicate name, rejecting enrollment - wazuh agent 4.3.4

### DISCUSSION THREAD

**Discussion by akuleckij** [wazuh/wazuh#14031] [Q&A] [answered]
Hello, I'm running wazuh-agent on CentOS 8.4, I'm getting an error Duplicate name, rejecting enrollment... when I try to install the agent for the first time.
My setup information:

Wazuh-manager:
Ubuntu 20.04.1
16GB RAM
AMD64 architecture (8 vCPU)

Wazuh-agent:
CentOS 8.4
2GB RAM
AMD64 architecture (1vCPU)

Manager logs:
2022/06/27 00:01:00 wazuh-authd: INFO: New connection from 10.175.0.5
2022/06/27 00:01:00 wazuh-authd: INFO: Received request for a new agent (centos-squid-proxy.novalocal) from: 10.175.0.5
2022/06/27 00:01:00 wazuh-authd: WARNING: Duplicate name 'centos-squid-proxy.novalocal', rejecting enrollment. Agent '017' key already exists on the manager.

Agent logs:
2022/06/27 09:35:49 wazuh-agentd: INFO: Requesting a key from server: 10.172.0.9
2022/06/27 09:35:49 wazuh-agentd: INFO: No authentication password provided
2022/06/27 09:35:49 wazuh-agentd: INFO: Using agent name as: centos-squid-proxy.novalocal
2022/06/27 09:35:49 wazuh-agentd: INFO: Waiting for server reply
2022/06/27 09:35:49 wazuh-agentd: ERROR: Duplicate agent name: centos-squid-proxy.novalocal (from manager)
2022/06/27 09:35:49 wazuh-agentd: ERROR: Unable to add agent (from manager)
2022/06/27 09:35:59 wazuh-agentd: WARNING: (4101): Waiting for server reply (not started). Tried: '10.172.0.9'.
2022/06/27 09:35:59 wazuh-agentd: WARNING: Unable to connect to any server.
2022/06/27 09:35:59 wazuh-agentd: INFO: Closing connection to server (10.172.0.9:1514/udp).
2022/06/27 09:35:59 wazuh-agentd: INFO: Trying to connect to server (10.172.0.9:1514/udp).

**Comment by xrisbarney** [comment, marked as answer]
Hi @akuleckij,

Thank you for using Wazuh!

Are you passing all your connections (including connections to the manager from agents) through the proxy (centos-squid-proxy.novalocal)?

I recommend the following:
- Edit your agent configuration file (/var/ossec/etc/ossec.conf) and specify a unique agent enrollment name in the enrollment section. A sample is below:
```
<client>
        <!--other pre existing configuration details -->
        <enrollment>
            <agent_name>centos-squid-proxy.novalocal.unique</agent_name>
        <!--other pre existing configuration details -->
        </enrollment>
    </client>
```
- The link to this documentation is [here](https://documentation.wazuh.com/current/user-manual/agent-enrollment/troubleshooting.html#troubleshooting-invalid-agent-name). 
- The link to ossec.conf agent options is [here](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/index.html).

Please let me know if this helps. Best!

**Reply by akuleckij** [reply]
Hi!
No, centos-squid-proxy.novalocal is the server on which the wazuh agent is installed. 
Judging by the logs, the agent has a unique name, do I still need to change something?
root@wazuh:/home/ubuntu# /var/ossec/bin/agent_control -l

Wazuh agent_control. List of available agents:
   ID: 000, Name: wazuh (server), IP: 127.0.0.1, Active/Local
   ID: 001, Name: w-akuletskij, IP: any, Active
   ID: 002, Name: dc1, IP: any, Active
   ID: 003, Name: ksc, IP: any, Active
   ID: 004, Name: 1c-srv, IP: any, Active
   ID: 010, Name: DC02, IP: any, Active
   ID: 017, Name: centos-squid-proxy.novalocal, IP: any, Never connected
   ID: 018, Name: W-iantipina, IP: any, Never connected

I don't find the duplicate names.
Although I tried anyway=) But nothing has changed.
root@wazuh:/home/ubuntu# /var/ossec/bin/agent_control -l

Wazuh agent_control. List of available agents:
   ID: 000, Name: wazuh (server), IP: 127.0.0.1, Active/Local
   ID: 001, Name: w-akuletskij, IP: any, Active
   ID: 002, Name: dc1, IP: any, Active
   ID: 003, Name: ksc, IP: any, Active
   ID: 004, Name: 1c-srv, IP: any, Active
   ID: 010, Name: DC02, IP: any, Active
   ID: 017, Name: centos-squid-proxy.novalocal, IP: any, Never connected
   ID: 018, Name: W-iantipina, IP: any, Never connected
   ID: 019, Name: centos-squid-proxy.novalocal.unique, IP: any, Never connected

root@wazuh:/home/ubuntu# cat /var/ossec/etc/client.keys
001 w-akuletskij any ab25a22eccb742fd87e118399eb199938c9dfe4affa71cf4f02d48c731fdb14e
002 dc-sbc any 23729e391fb06b3993ef2ac38d2b2bbce0dc465879eb4ce51b02c3dc7c97752a
003 ksc-sberauto any fbcd5f0dfaf5e6839c5843d149dd2b5665323b7eb3c6722b08df1c79c694d170
004 1c-srv any 5984b96a1fbc8b492292f473d655fa40c89f27434909f2a76ff8ee15cc01574c
010 DC02MosLen any 27e2031c1ecc181a2387726da74ed814c3e8a91c8f86bc43698cbdab19415da7
017 centos-squid-proxy.novalocal any feb089f51d4a0d9ebc3fdfafee04170672ae8e2bd6588d7912dadf664de15ce3
018 W-iantipina any ec58b90c2774adb9ab9add0e39894ed4326cc773a0f13c3f5a5f2a3a64ce9813
019 centos-squid-proxy.novalocal.unique any 2f36a6596e48ecc47ddfd1d74b9987625fefdc7ce8afe7af150035c047f872ee

root@wazuh:/home/ubuntu# ls -lh /var/ossec/queue/db/
total 166M
-rw-r----- 1 wazuh wazuh 2.2M Jun 27 08:33 000.db
-rw-r----- 1 wazuh wazuh  40M Jun 27 08:37 001.db
-rw-r----- 1 wazuh wazuh  34M Jun 27 08:37 002.db
-rw-r----- 1 wazuh wazuh  30M Jun 27 08:33 003.db
-rw-r----- 1 wazuh wazuh  31M Jun 27 08:34 004.db
-rw-r----- 1 wazuh wazuh  31M Jun 27 08:37 010.db
-rw-r----- 1 wazuh wazuh  56K Jun 27 08:37 global.db
-rw-r----- 1 wazuh wazuh  17K Jun 27 08:37 global.db-journal
srw-rw---- 1 wazuh wazuh    0 Jun 20 10:30 wdb


I will be grateful for any help.

**Reply by akuleckij** [reply]
![image](https://user-images.githubusercontent.com/108254972/175897926-ca406c16-99e0-4c22-bd0a-a1deda327535.png)

**Reply by akuleckij** [reply]
The same thing happens on Windows computers (W-iantipina). At the same time, if you look at the traffic using tcpdump, then on the manager it is from agents. Agents have full access to manager. And there are computers that connect normally (status active).

**Reply by xrisbarney** [reply]
please try to restart the agent and let me know if it connects:

`systemctl restart wazuh-agent`

Also, what version of Wazuh manager and Wazuh agent are you running? v4.3.2 had a bug that was [duplicating the enrollment tags](https://github.com/wazuh/wazuh/releases/tag/v4.3.3).

- What command are you using to install and enroll the agent?
- Can you share your agent ossec.conf for the problematic machine? (Please remove all sensitive info)

**Reply by akuleckij** [reply]
I did, but nothing has changed.

**Reply by xrisbarney** [reply]
Okay.

Also, what version of Wazuh manager and Wazuh agent are you running? v4.3.2 had a bug that was [duplicating the enrollment tags](https://github.com/wazuh/wazuh/releases/tag/v4.3.3).

- What command are you using to install and enroll the agent?
- Please share your agent ossec.conf for the problematic machines? (Please remove all sensitive info)

**Reply by akuleckij** [reply]
Wazuh manager - v4.3.4
![image](https://user-images.githubusercontent.com/108254972/175914633-e782a05e-9e80-4dd7-bc95-795b41de13eb.png)

**Reply by akuleckij** [reply]
Wazuh agent - 4.3.4-1
sudo WAZUH_MANAGER='10.172.0.9' WAZUH_PROTOCOL='UDP' WAZUH_AGENT_GROUP='default' yum install https://packages.wazuh.com/4.x/yum/wazuh-agent-4.3.4-1.x86_64.rpm

sudo systemctl daemon-reload
sudo systemctl enable wazuh-agent
sudo systemctl start wazuh-agent

10.172.0.9 - IP address WAZUH MANAGER srever

After executing these commands on other computers, everything was fine.

**Reply by akuleckij** [reply]
[ossec.txt](https://github.com/wazuh/wazuh/files/8990910/ossec.txt)

**Reply by xrisbarney** [reply]
Thank you for providing this information. 

Your ossec.conf file looks fine, the configuration process is also okay. I have updated the ossec.conf file below with the following:
- I changed the connection method from UDP to TCP.
- I added the block `<manager_address>10.172.0.9</manager_address>` please replace the Centos ossec.conf with the contents of this one, restart the agent and let me know if the agent connects.

[ossec (1).txt](https://github.com/wazuh/wazuh/files/8991010/ossec.1.txt)

Also, check the agent logs and send the error logs you see after the agent restarts.

**Reply by akuleckij** [reply]
Super!!!
This helped both for linux computers and for Windows. Now everything works great!
Thank you very much!!!

**Reply by xrisbarney** [reply]
Glad to have been able to help!
